lib/manager/components/version/FeedVersionSpanChart.js

```diff
@@ -0,0 +1,266 @@
+        </div>
+      </div>
+    )
+  }
+}
+
+/**
+ * Compute label position and sizing info.
+ */
+function getRenderVars ({
+  firstDate,
+  lastDate,
+  versionStartDate,
+  versionEndDate,
+  versionLabel
+}) {
+  const totalDays = lastDate.diff(firstDate, 'days')
+  const daysActive = versionEndDate.diff(versionStartDate, 'days')
+  const versionOffset = versionStartDate.diff(firstDate, 'days')
+
+  return {
+    daysActive,
+    endLabel: versionEndDate.format(DATE_FORMAT),
+    relativeLength: daysActive / totalDays,
+    relativeOffset: versionOffset / totalDays,
+    startLabel: versionStartDate.format(DATE_FORMAT),
+    totalDays,
+    versionOffset
+  }
+}
+
+/** Renders the validity span of one feed using the given props. */
+const FeedSpan = props => {
+  const { isServiceGap = false, top } = props
+  const { daysActive, relativeLength, relativeOffset } = getRenderVars(props)
+  return (
+    <OverlayTrigger
+      placement='bottom'
+      overlay={<Tooltip id={`feed-span`}>
+        {isServiceGap && 'No service for'}{' '}
+        {daysActive} days
+      </Tooltip>}>
+      <span
+        style={{
+          height: HEIGHT_PROP,
+          left: `${relativeOffset * BAR_WIDTH_PX + SPAN_MARGIN_PX + DATE_WIDTH_PX}px`,
+          position: 'absolute',
+          top,
+          width: `${relativeLength * BAR_WIDTH_PX}px`
+        }}>
+        <span
+          style={{
+            backgroundColor: isServiceGap ? 'red' : 'grey',
+            height: '.75em',
+            position: 'absolute',
+            bottom: '10%',
+            width: '100%'
+          }}
+        />
+      </span>
+    </OverlayTrigger>
+  )
+}
+
+/**
+ * Renders a validity span START label (needs to be separate from the span itself for Z order).
+ */
+const FeedStartLabel = props => {
+  const { top } = props
+  const { relativeOffset, startLabel, totalDays, versionOffset } = getRenderVars(props)
+  return (
+    <span
+      style={{
+        left: `${relativeOffset * BAR_WIDTH_PX}px`,
+        position: 'absolute',
+        textAlign: 'right',
+        top,
+        width: `${DATE_WIDTH_PX}px`,
+        zIndex: DATE_LABEL_ZINDEX
+      }}>
+      {startLabel}
+    </span>
+  )
+}
+
+/**
+ * Renders a validity span END label (needs to be separate from the span itself for Z order).
+ */
+const FeedEndLabel = props => {
+  const { top, versionLabel } = props
+  const {
+    daysActive,
+    endLabel,
+    relativeLength,
+    relativeOffset,
+    totalDays,
+    versionOffset
+  } = getRenderVars(props)
+
+  return (
+    <span
+      style={{
+        left: `${relativeOffset * BAR_WIDTH_PX + SPAN_MARGIN_PX + DATE_WIDTH_PX + relativeLength * BAR_WIDTH_PX + SPAN_MARGIN_PX}px`,
+        position: 'absolute',
+        top,
+        whiteSpace: 'nowrap',
+        zIndex: DATE_LABEL_ZINDEX
+      }}>
+      {endLabel} {versionLabel}
+    </span>
+  )
+}
+
+/** Renders a marker on today's date (if one of the feeds being compared is valid). */
+const TodayMarker = ({
```

Comment:
I think it'd be nice to have more than just a bar that has an OverlayTrigger. It'd be great to have some kind of label beneath this saying that it marks the current date.